### ISSUE TITLE
Macro incorrectly parsed

### ISSUE THREAD

**Issue by mrx23dot** [microsoft/vscode-cpptools#12326]
Type: <b>Bug</b>

vscode parses this differently from GCC, both should be active
![image](https://github.com/microsoft/vscode-cpptools/assets/4163396/f771ebe7-21b6-4f25-b00d-ec2e9e740c62)

```
#define XXBBX_ENABLED 1u
#define NRF_MODULE_ENABLED(module)   defined(module ## _ENABLED) 
 
#if XXBBX_ENABLED
active(); //ok
#endif

#if NRF_MODULE_ENABLED(XXBBX)
inactive(); //nok
#endif
```

Extension version: 1.17.5
VS Code version: Code 1.73.1 (6261075646f055b99068d3688932416f2346dd3b, 2022-11-09T04:27:29.066Z)
OS version: Windows_NT x64 10.0.19045
Modes: Unsupported
Sandboxed: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-10510U CPU @ 1.80GHz (8 x 2304)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|undefined|
|Memory (System)|15.81GB (5.38GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details>

**Comment by browntarik** [comment]
Could explain what the expected behavior shood be? Are you saying that the below code should not be highlighted as inactive?

#if NRF_MODULE_ENABLED(XXBBX)
inactive(); //nok
#endif

**Comment by browntarik** [comment]
Thank you for reporting this issue, this has been reported to our internal team (VS: 2078589) and we will provide an update when it has been addressed. Please keep track of this issue to receive further updates.

**Comment by sean-mcmanus** [comment]
@mrx23dot Yes, the fix won't involve an increase in the VS Code min version, so Windows 7 support won't be dropped, but it's possible some other fix may eventually require us to need to drop Windows 7.

**Comment by simon-r** [comment]
I've the same problem:

When I open the file  the source code is highlighted correctly.
But when a switch the value of the macro the highlighting is not automatically  updated.

![immagine](https://github.com/user-attachments/assets/f82b7b59-7f89-488d-9527-c7954b308dfc)


Extension version: 1.21.6
vscode version:  1.92.2
Operating System: openSUSE Leap 15.6

System:

> KDE Plasma Version: 5.27.11
> KDE Frameworks Version: 5.115.0
> Qt Version: 5.15.12
> Kernel Version: 6.4.0-150600.23.17-default (64-bit)
> Graphics Platform: X11
> Processors: 36 × Intel® Core™ i9-10980XE CPU @ 3.00GHz
> Memory: 125.5 GiB of RAM
> Graphics Processor: NVIDIA GeForce RTX 2060/PCIe/SSE2
